deploy/v1.8.0/avalanche.ts

```diff
@@ -0,0 +1,94 @@
+ *
+ * NOTE: The final step to complete the update is a transition of the manager role on a PrizeDistributionBuffer to be the newly deployed PrizeDistributionFactoryV2.
+ */
+export default async function deployToAvalancheMainnet(hre: HardhatRuntimeEnvironment) {
+  if (process.env.DEPLOY === 'v1.8.0.avalanche') {
+    dim(`Deploying: PrizeTierHistoryV2 and PrizeDistributionFactoryV2 on Avalanche Mainnet`);
+    dim(`Version: 1.8.0`);
+  } else {
+    return;
+  }
+
+  const { getNamedAccounts, ethers, getChainId } = hre;
+
+  const { deployer, executiveTeam, defenderRelayer } = await getNamedAccounts();
+
+  const chainId = parseInt(await getChainId(), 10);
+
+  dim(`chainId ${chainId} `);
+  dim(`---------------------------------------------------`);
+  dim(`Named Accounts`);
+  dim(`---------------------------------------------------`);
+  dim(`deployer: ${deployer}`);
+  dim(`executiveTeam: ${executiveTeam}`);
+  dim(`defenderRelayer: ${defenderRelayer}`);
+  dim(`---------------------------------------------------\n`);
+  const startingBalance = await ethers.provider.getBalance((await ethers.getSigners())[0].address);
+
+  // ===================================================
+  // Deploy Contracts
+  // ===================================================
+
+  // Load existing contracts
+  const ticket = await hre.deployments.get('Ticket');
+  const pdb = await hre.deployments.get('PrizeDistributionBuffer');
+  const db = await hre.deployments.get('DrawBuffer');
+
+  // 1. Deploy or load PrizeTierHistoryV2
+  const pthv2 = await deployAndLog('PrizeTierHistoryV2', {
+    from: deployer,
+    args: [deployer],
+    skipIfAlreadyDeployed: true,
+  });
+
+  // 2. Deploy or load PrizeDistributionFactoryV2
+  const pdfv2 = await deployAndLog('PrizeDistributionFactoryV2', {
+    from: deployer,
+    args: [
+      deployer,
+      pthv2.address,
+      db.address,
+      pdb.address,
+      ticket.address,
+      PRIZE_DISTRIBUTION_FACTORY_MINIMUM_PICK_COST,
+    ],
+    skipIfAlreadyDeployed: true,
+  });
+
+  // ===================================================
+  // Configure Contracts
+  // ===================================================
+
+  // 1. Set Manager of PTHV2 to Executive Team
+  await setManager('PrizeTierHistoryV2', null, executiveTeam);
+
+  // 2. Set Manager of PDFV2 to Defender Relayer
+  await setManager('PrizeDistributionFactoryV2', null, defenderRelayer);
+
+  // 3. Transfer Ownership of PTHV2 to Executive Team
+  await transferOwnership('PrizeTierHistoryV2', null, executiveTeam);
+
+  // 4. Transfer Ownership of PDFV2 to Executive Team
+  await transferOwnership('PrizeDistributionFactoryV2', null, executiveTeam);
+
+  dim(`---------------------------------------------------`);
+  green(
+    `NOTE: The final step to complete the update is a transition of the manager role on the PrizeDistributionBuffer at ${pdb.address} to be the newly deployed PrizeDistributionFactoryV2.`,
```

Comment:
And don't forget to update the PrizeDistributor to call the DrawCalculator directly and not the DrawCalculatorTimelock: `prizeDistributor.setDrawCalculator(drawCalculatorAddress)`
⇒ it removes the DrawCalculatorTimelock from the architecture